src/Gumps/GumpResourceTracker.cpp

```diff
@@ -0,0 +1,329 @@
+    : CGump(GT_RESOURCETRACKER, 0, x, y)
+{
+    m_Background = (CGUIResizepic *)Add(new CGUIResizepic(0, 0xA3C, 0, 0, Width, Height));
+    m_Trans = (CGUIChecktrans *)Add(new CGUIChecktrans(5, 5, 505, 65));
+    m_gridDataBoxGUI = (CGUIDataBox *)Add(new CGUIDataBox());
+    m_itemsDataBoxGui = (CGUIDataBox *)Add(new CGUIDataBox());
+    numRow = Height / boxDimension;
+    numCol = Width / boxDimension;
+    DrawGrid();
+    DrawItem();
+
+    m_Resizer = (CGUIResizeButton *)Add(
+        new CGUIResizeButton(ID_GWM_RESIZE, 0x0837, 0x0838, 0x0838, Width - 5, Height - 5));
+}
+
+CGumpResourceTracker::~CGumpResourceTracker()
+{
+}
+
+void CGumpResourceTracker::PrepareContent()
+{
+    CPoint2Di offset = g_MouseManager.RealPosition;
+    offset.X -= GetX();
+    offset.Y -= GetY();
+    currCol = offset.X / boxDimension;
+    currRow = offset.Y / boxDimension;
+
+    if (currCol > numCol - 1)
+    {
+        currCol = -1;
+    }
+    if (currRow > numRow - 1)
+    {
+        currRow = -1;
+    }
+
+    if (g_SelectedObject.Gump == this && g_ObjectInHand.Enabled && offset.X > 0 && offset.Y > 0)
+    {
+        hasItemInGump = true;
+        drawDebug = true;
+        WantUpdateContent = true;
+    }
+    else
+    {
+        drawDebug = false;
+    }
+}
+
+void CGumpResourceTracker::UpdateContent()
+{
+    CGump::UpdateContent();
+    WantRedraw = true;
+}
+
+void CGumpResourceTracker::Draw()
+{
+    char dbf[150] = { 0 };
+    if (g_DeveloperMode == DM_DEBUGGING && drawDebug)
+    {
+        sprintf_s(
+            dbf, //column row
+            "Curent selection :\nColumn=%i Row=%i\nObject in hand=0x%04X",
+            currCol,
+            currRow,
+            g_ObjectInHand.Graphic);
+        g_FontManager.DrawA(3, dbf, 0x35, GetX(), GetY() - 70);
+    }
+    CGump::Draw();
+}
+
+void CGumpResourceTracker::DrawText()
+{
+    for (uint8_t r = 0; r < numRow; r++)
+    {
+        for (uint8_t c = 0; c < numCol; c++)
+        {
+            uint16_t graph = items[r][c].graphic;
+            if (graph != 0)
+            {
+                int count = CountItemBackPack(graph);
+                CGUIText *txt = new CGUIText(
+                    0x44, (c * boxDimension) + 5, (r * boxDimension) + boxDimension - 15);
+                if (count <= 20 && count > 10)
+                {
+                    txt->Color = 0x99; //yellow
+                }
+                else if (count <= 10)
+                {
+                    txt->Color = 0x26; //red
+                }
+                else
+                {
+                    txt->Color = 0x44;
+                }
+                txt->CreateTextureA(9, str_from(count), boxDimension, TS_CENTER);
+                m_itemsDataBoxGui->Add(txt);
+            }
+        }
+    }
+}
+
+void CGumpResourceTracker::DrawItem()
+{
+    for (uint8_t r = 0; r < numRow; r++)
+    {
+        for (uint8_t c = 0; c < numCol; c++)
+        {
+            TrackedItem item = items[r][c];
```

Comment:
here you're copying data, the can become performance expensive as code change.
Use a reference to the data instead, also const as you're not modifying it:
`const auto &item = items[r][c];`